From 1c1363586d0e60bba1e18fbff58eb4baf9c1d61b Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@kdab.com>
Date: Mon, 19 Apr 2021 09:42:41 +0200
Subject: [PATCH 19/19] QPageSize: make PageSizeId ctor non-explicit

A QPageSize::PageSizeId is a faithful representation of a QPageSize,
so the corresponding QPageSize ctor shouldn't be explicit.

[ChangeLog][QtGui][QPageSize] Conversion from a QPageSize::PageSizeId
is now implicit.

Change-Id: I2d32da370c032949686757400cb7c28583d9d8ac
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
Reviewed-by: Giuseppe D'Angelo <giuseppe.dangelo@kdab.com>
(cherry picked from commit c8f380bd13f077cd797edbdb55723a2524f55c78)
Reviewed-by: Qt CI Bot <qt_ci_bot@qt-project.org>
---
 qtbase/src/gui/painting/qpagesize.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qtbase/src/gui/painting/qpagesize.h b/qtbase/src/gui/painting/qpagesize.h
index 133274760f..5b53eae458 100644
--- a/qtbase/src/gui/painting/qpagesize.h
+++ b/qtbase/src/gui/painting/qpagesize.h
@@ -228,7 +228,7 @@ public:
     };
 
     QPageSize();
-    explicit QPageSize(PageSizeId pageSizeId);
+    /*implicit*/ QPageSize(PageSizeId pageSizeId);
     explicit QPageSize(const QSize &pointSize,
                        const QString &name = QString(),
                        SizeMatchPolicy matchPolicy = FuzzyMatch);
-- 
2.44.0

